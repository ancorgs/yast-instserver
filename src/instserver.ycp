/**
 * File:	clients/instserver.ycp
 * Package:	Configuration of instserver
 * Summary:	Main file
 * Authors:	Anas Nashif <nashif@suse.de>
 *
 * $Id$
 *
 * Main file for instserver configuration. Uses all other files.
 */

{

/***
 * <h3>Configuration of instserver</h3>
 */

textdomain "instserver";

/* The main () */
y2milestone ("----------------------------------------");
y2milestone ("Instserver module started");

import "Progress";
import "Report";
import "Summary";

import "CommandLine";
include "instserver/wizards.ycp";

map cmdline_description = $[
    "id" 	: "instserver",
    /* Command line help text for the Xinstserver module */
    "help"	: _("Configuration of instserver"),
    "guihandler"        : InstserverSequence,
    "initialize"        : Instserver::Read,
    "finish"            : Instserver::Write,
    "actions"           : $[
	// FIXME TODO: fill the functionality description here
    ],
    "options"		: $[
	// FIXME TODO: fill the option descriptions here
    ],
    "mapping"		: $[
	// FIXME TODO: fill the mappings of actions and options here
    ]
];

/* is this proposal or not? */
boolean propose = false;
list args = WFM::Args();
if(size(args) > 0) {
    if(is(WFM::Args(0), path) && WFM::Args(0) == .propose) {
        y2milestone("Using PROPOSE mode");
        propose = true;
    }
}

/* main ui function */
any ret = nil;

if(propose) ret = InstserverAutoSequence();
else ret = CommandLine::Run(cmdline_description);
y2debug("ret=%1", ret);

/* Finish */
y2milestone("Instserver module finished");
y2milestone("----------------------------------------");

return ret;

/* EOF */
}
