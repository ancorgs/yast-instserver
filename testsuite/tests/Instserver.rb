# encoding: utf-8

module Yast
  class InstserverClient < Client
    def main
      # testedfiles: Instserver.ycp

      Yast.include self, "testsuite.rb"
      TESTSUITE_INIT([], nil)

      Yast.import "Instserver"

      DUMP("Instserver::Modified")
      TEST(lambda { Instserver.Modified }, [], nil)

      DUMP("Instserver::EscapeSLPData()")
      TEST(lambda do
        Instserver.EscapeSLPData(
          { "description" => "No need to escpace anything." }
        )
      end, [], nil)
      TEST(lambda do
        Instserver.EscapeSLPData(
          {
            "description" => "Rounded brackets (), \\ backslash and comma , must be escpaced."
          }
        )
      end, [], nil)
      TEST(lambda do
        Instserver.EscapeSLPData({ "escaped.=name" => "value\\\\\\." })
      end, [], nil)

      nil
    end
  end
end

Yast::InstserverClient.new.main
