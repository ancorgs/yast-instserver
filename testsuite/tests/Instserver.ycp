{

// testedfiles: Instserver.ycp

include "testsuite.ycp";
TESTSUITE_INIT([], nil);

import "Instserver";

DUMP("Instserver::Modified");
TEST(``(Instserver::Modified()), [], nil);

}
