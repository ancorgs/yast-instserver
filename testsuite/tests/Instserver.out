Dump	Instserver::Modified
Return	false
